lib/stat_tracker.rb

```diff
@@ -166,117 +166,84 @@ def average_goals_by_season
     game_count_by_team_id = Hash.new(0)
-    games_csv.each do |row|
-      away_goals_by_team_id[row[:away_team_id].to_i] += row[:away_goals].to_i
-      game_count_by_team_id[row[:away_team_id].to_i] += 1
+    games.each do |game|
+      away_goals_by_team_id[game.away_team_id.to_i] += game.away_goals.to_i
+      game_count_by_team_id[game.away_team_id.to_i] += 1
     end
     average_teamid_score = away_goals_by_team_id.map { |team_id, goals| [team_id, goals.to_f / game_count_by_team_id[team_id]] }.to_h
     best_away_team_id = average_teamid_score.max_by { |away_team_id, average_goals| average_goals }[0]
-    teams_csv = CSV.open(@team_path, headers: true, header_converters: :symbol)
-    visitor_team_row = teams_csv.find do |row|
-      row[:team_id].to_i == best_away_team_id
-    end
-    visitor_team_row[:teamname]
+    team_info(best_away_team_id.to_s)['team_name']
   end
 
   def highest_scoring_home_team
-    games_csv = CSV.open(@game_path, headers: true, header_converters: :symbol)
     home_goals_by_team_id = Hash.new(0)
     game_count_by_team_id = Hash.new(0)
-    games_csv.each do |row|
-      home_goals_by_team_id[row[:home_team_id].to_i] += row[:home_goals].to_i
-      game_count_by_team_id[row[:home_team_id].to_i] += 1
+    games.each do |game|
+      home_goals_by_team_id[game.home_team_id.to_i] += game.home_goals.to_i
+      game_count_by_team_id[game.home_team_id.to_i] += 1
     end
     average_teamid_score = home_goals_by_team_id.map { |team_id, goals| [team_id, goals.to_f / game_count_by_team_id[team_id]] }.to_h
     best_home_team_id = average_teamid_score.max_by { |home_team_id, average_goals| average_goals }[0]
-    teams_csv = CSV.open(@team_path, headers: true, header_converters: :symbol)
-    visitor_team_row = teams_csv.find do |row|
-      row[:team_id].to_i == best_home_team_id
-    end
-    visitor_team_row[:teamname]
+    team_info(best_home_team_id.to_s)['team_name']
   end
 
   def lowest_scoring_visitor
-    games_csv = CSV.open(@game_path, headers: true, header_converters: :symbol)
     away_goals_by_team_id = Hash.new(0)
     game_count_by_team_id = Hash.new(0)
-    games_csv.each do |row|
-      away_goals_by_team_id[row[:away_team_id].to_i] += row[:away_goals].to_i
-      game_count_by_team_id[row[:away_team_id].to_i] += 1
+    games.each do |game|
+      away_goals_by_team_id[game.away_team_id.to_i] += game.away_goals.to_i
+      game_count_by_team_id[game.away_team_id.to_i] += 1
     end
     average_teamid_score = away_goals_by_team_id.map { |team_id, goals| [team_id, goals.to_f / game_count_by_team_id[team_id]] }.to_h
     worst_away_team_id = average_teamid_score.min_by { |away_team_id, average_goals| average_goals }[0]
-    teams_csv = CSV.open(@team_path, headers: true, header_converters: :symbol)
-    visitor_team_row = teams_csv.find do |row|
-      row[:team_id].to_i == worst_away_team_id
-    end
-    visitor_team_row[:teamname]
+    team_info(worst_away_team_id.to_s)['team_name']
```

Comment:
excellent use of team_info. this is going to be really useful in all our other methods